### PR TITLE
Make the class-loader cache build-time and add a loader debug page

### LINKED ISSUE #30: Autoloader cache not refreshing
### Describe the bug

I found that when the `class-loader-cache` is created, it isn't refreshing when new (or existing) classes are created/changed. I found instances where my cache was over 2 weeks old despite many changes performed to related classes.

My solve for this was to SFTP into my WP Engine hosted server and manually delete the `class-loader-cache` directory to ensure it grabs the latest changes.

This results in scenarios of "works locally but not on my servers". I think this issue stems out of iterative deployments. One scenario in my testing, it worked just fine, but it was a server that we hadn't deployed any of the work to yet before any caches were created. If a cache exists and you adjust something in the autoloaded classes, the cache doesn't pick those up.

Ideally there is a mechanism to manually refresh that cache or it auto refreshes when it detects that a class has been modified. [I did see the const to skip cache](https://github.com/10up/wp-framework/pull/26) which will be really helpful in debugging this in the future, but ideally we have a reliable source to auto refresh it along with a manual refresh option.

### Steps to Reproduce

1. Create a new class using the ModuleInterface and deploy to your server.
2. Make some changes that you can observe either from the front or backend (both affected) and deploy.
3. Double check if your changes can be observed and double check the last modified date of the `class-loader-cache` and your autoloaded classes.
4. If the changes aren't showing up due to the cache, deleting the `class-loader-cache` will fix the issue.

### Screenshots, screen recording, code snippet

Image
Observe the last modified dates not in alignment. This reflects the effect of the site working in a past state despite all the files being up to date.

### Environment information

Myself and another co-worker reported this issue on WP Engine hosted sites so unsure this is relevant at all. There is a thread in the company slack around this.

### WordPress information

WordPress 6.8.3

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by darylldoyle**
## Summary

Reworks the class-loader cache so it is generated at build time and only ever read at
runtime, and adds a hidden admin page for inspecting loader caches on a running site.

Closes #30.

## Screenshots

_Loader debug page — `admin.php?page=tenup-framework-loaders`:_

CleanShot 2026-06-29 at 09 58 57

## ⚠️ Breaking change (targets a major release)

The runtime no longer generates the cache automatically. On 1.x the first
production/staging request wrote the cache and later requests read it forever — the
stale-cache bug behind #30. Now the framework **reads** a pre-built cache if present and
**discovers live** otherwise; it never writes. Caching becomes opt-in via a build step.

Existing installs that upgrade without adding the build step keep working but run
uncached. See the new [Upgrade Guide](docs/Upgrade-Guide.md).

Removed: `should_use_cache()`, the production/staging gating, and `VIP_GO_APP_ENVIRONMENT`
handling. `TENUP_FRAMEWORK_DISABLE_CLASS_CACHE` now means "ignore any shipped cache and
discover live." Cache filename bumped to `class-loader-cache-v2.php` so 1.x caches are
ignored after upgrade.

## What's included

**Build-time cache**
- `ReadOnlyFileDiscoverCacheDriver` — read-only Spatie driver (no-op writes, no `mkdir`).
- `ModuleInitialization::generate_cache()` plus `bin/tenup-framework-generate-class-cache`
  (runs without bootstrapping WordPress) and a `composer generate-class-cache` alias.

**Loader debug page**
- Hidden, admin-only page at `admin.php?page=tenup-framework-loaders` (`manage_options`).
- Aggregates every loader cache across all framework copies via the
  `tenup_framework_debug_loaders` filter (works across php-scoped / multi-version copies).
- Per loader: a colour-coded status badge, owner, cache path/detail, legacy-file warnings,
  a collapsible list of loaded classes (with resolved files), and an on-demand
  live-vs-cache staleness check.
- Recording and the page are gated behind `is_admin()`, so the front end loads none of it.
  Read-only; disable via `tenup_framework_enable_loader_debug` / `TENUP_FRAMEWORK_DISABLE_LOADER_DEBUG`.

## Docs
- New: `Build-and-Deployment.md` (CLI + GitHub Actions/GitLab CI/CircleCI examples + per-package
  model), `Debugging.md`, `Upgrade-Guide.md`.
- Rewrote the cache sections of `Autoloading.md` and `Modules-and-Initialization.md`.

## Testing
- `composer lint` (PHPCS 10up), `composer static` (PHPStan level 10), and `composer test`
  (63 tests) all pass.
- Coverage includes the read-only driver, generate/read paths, admin vs front-end dispatch,
  the debug registry/page, and the staleness diff.

## Credits

Props @darylldoyle, @rleeson. Build-command tests, discovery/lookup timing, and the corrupt-cache runtime resilience came from #36 (@rleeson), merged into this branch; the debug page then surfaces a failed cache load rather than reporting it healthy.

### CODE REVIEW

**Comment by claytoncollie** [comment]
@darylldoyle This is very nice and thorough. I read through all of the information and want to make sure that the generate cache script can run at the project root and build individual caches for each of the themes and plugins using wp-framework. Or do we have to run the script multiple times if the theme and a plugin both use WPF?

**Comment by rleeson** [comment]
@darylldoyle The Debug page is a great addition. I validated the changes are working on a project where the framework is used in 5 separate instances, incrementally turning them on and verifying functionality. There were a few improvements to testing and the debug display which I worked into a PR (#36) branched from this PR.

File: src/ModuleInitialization.php [hunk 1]
```diff
@@ -9,10 +9,13 @@
 
 namespace TenupFramework;
 
+use Composer\InstalledVersions;
 use ReflectionClass;
 use Spatie\StructureDiscoverer\Cache\FileDiscoverCacheDriver;
 use Spatie\StructureDiscoverer\Data\DiscoveredStructure;
```
Comment (rleeson) [inline src/ModuleInitialization.php]: No longer in use, can be removed

File: src/Debug/LoaderDebug.php [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+	}
+
+	/**
+	 * Output the page's scoped styles once.
+	 *
+	 * @return void
+	 */
+	protected static function render_styles() {
+		echo '<style>
+			.tenup-loaders .tenup-loader { max-width: 60em; margin: 1.25em 0; padding: .5em 1.25em 1.25em; background: #fff; border: 1px solid #c3c4c7; border-radius: 4px; }
```
Comment (rleeson) [inline src/Debug/LoaderDebug.php]: The expanded class list can overflow the containing box boundary depending on word wrapping. This should be adjusted to fit the content.

**Review by rleeson — CHANGES_REQUESTED** [review]
@darylldoyle I've submitted a PR (#36) with some additions I think improve both test coverage and debug capabilities.

The adjustments I would consider if those changes are not used, I'll mark inline.

**Comment by darylldoyle** [comment]
Noting that if a cache file is corrupted, it will now show like this:

CleanShot 2026-07-30 at 16 08 09

**Comment by rleeson** [comment]
@darylldoyle I added another stacked PR (#39) for a build edge case and some documentation and test adjustments. Other than those, my project testing looks good.
